tests/test_appelb.py

```diff
@@ -364,6 +364,30 @@ def test_appelb_delete_force(self):
         resources = p.run()
         self.assertEqual(len(resources), 1)
 
+    def test_appelb_modify_attributes(self):
+        session_factory = self.replay_flight_data("test_appelb_modify_attributes")
+        client = session_factory().client("elbv2")
+        p = self.load_policy(
+            {
+                "name": "appelb-enable-deletion-protection",
+                "resource": "app-elb",
+                "actions": [
+                    {
```

Comment:
ideally we want a filter to validate that the state of the filter before the action was that it didn't have delete protection enabled.

Reply:
Uh does that mean that I have to implement a new filter based on `describe_load_balancer_attributes` or some such? :-)

Reply:
Yeah it looks like it would need the describe_load_balancer_attributes call to pull back the info to see if the given setting is already enabled/disabled.  Currently custodian doesn't pull back or show if Delete Protection is enabled for app-elbs

Reply:
yeah we it looks like we need a new filter as well

Reply:
Is there a way to leverage `ValueFilter` or something similar for this? Such that my new `app-elb` based filter first extracts the attributes via `describe_load_balancer_attributes` but then delegates to `ValueFilter` or equivalent, to do the actual work, with the goal of making this consistent with other stuff as well as flexible? The alternative I think is to build a filter that just matches == on the load balancer attributes, and it would look very much like the action in this PR, but not support any sort of more fancy filterting that maybe `ValueFilter` could give us out of the box? What's the right thing to do here?

Reply:
Send help. I'm starting to talk to myself. :-D

But so I figured out how it could be done one way hooking up `ValueFilter`, namely via deriving and overriding `ValueFilter.get_resource_value`. Worthwhile or nah?

Reply:
definitely its very common to subclass value filter and just override in get_resource_value or more commonly override process, enrich via annotation (so multiple attributes can be filtered without extra api calls), and then call super with get_resource_value override to just return the annotation.

Reply:
As per my last commit message:

Hah, I guess I didn't see that we already had
c7n.resources.appelb.CheckAttributes.  Oh well.  The only thing that's
maybe not great here is that the ValueFilter based CheckAttributes
expects the value to be a bool (a yaml `true`), while the attributes
modifier likes it to be a literal `true` as a string.

Reply:
ideally we'd match syntax on the filter and support marshalling for the action

Reply:
i pushed a fix for this so it supports True, False, or strings, also fixed an issue with passing numbers.